### PR TITLE
query param restrictions

### PR DESCRIPTION
**Pull request by tkellen**

### CODE REVIEW

**Comment by tkellen** [comment]
I'd like to allow camel casing in other languages too.  Does anyone know how to classify uppercase characters in that way?

**Comment by tkellen** [comment]
Discussed this with @dgeb and backed into it by requiring any non a-z character in custom query params.

File: format/index.md [hunk 1]
```diff
@@ -1732,6 +1732,17 @@ A server **MUST** prepare responses, and a client **MUST** interpret
 responses, in accordance with
 [`HTTP semantics`](http://tools.ietf.org/html/rfc7231).
 
+## Query Parameters <a href="#query-parameters" id="query-parameters" class="headerlink"></a>
+
+Implementation specific query parameters **MUST** adhere to the same constraints
+as [member names] with the exception that they **MUST** contain at least one
```
Comment (dgeb) [inline format/index.md]: perhaps "additional requirement" instead of "exception"?

**Comment by dgeb** [comment]
Thanks @tkellen :shipit:

**Comment by ethanresnick** [comment]
:+1:

**Pull request merged**